EngineBay.DemoModule/Dtos/TodoItemDto.cs

```diff
@@ -0,0 +1,37 @@
+namespace EngineBay.DemoModule
+{
+    public class TodoItemDto
+    {
+        public TodoItemDto(TodoItem todoItem)
+        {
+            if (todoItem is null)
+            {
+                throw new ArgumentNullException(nameof(todoItem));
+            }
+
+            this.Name = todoItem.Name;
+            this.ListId = todoItem.ListId;
+            this.List = todoItem.List;
+            this.Completed = todoItem.Completed;
+            this.Description = todoItem.Description;
+            this.CompletedDate = todoItem.CompletedDate;
+            this.DueDate = todoItem.DueDate;
+        }
+
+        public Guid Id { get; set; }
+
+        public string Name { get; set; }
+
+        public Guid ListId { get; set; }
+
+        public virtual TodoList? List { get; set; }
```

Comment:
virtual?

Reply:
I saw it being done with other models that had foreign keys (see BasicAuthCredential), and when I looked up why that could be done - I found that it can be used for lazy loading. https://learn.microsoft.com/en-us/ef/core/querying/related-data/lazy

Reply:
but this is a dto?

Reply:
Oops